language/move-prover/interpreter/src/concrete/ty.rs

```diff
@@ -4,6 +4,7 @@
 use std::fmt;
 
```

Comment:
Could you add a //! comment why this is needed and what's the difference to the existing `ty.rs`? And do we actually need it? I do not see an obvious diff besides you expand names and field info, but all this info can be already obtained via the model.

I'm concerned about the additional maintenance burden when we extend the type system.

If you need additional types for the interpreter, perhaps we can extend the model type?

Reply:
Yes, I share the concern on the extra maintenance burden. The type system for the interpreter is a strict (and very restricted) subset of what is offered in the move-model. My personal preference will be to have it stay in a very restricted way, not only because we don't need those extra types for interpretation (and hence, save a couple of `_ => unreachable!()`, but also because this enables me to write code in a more precise way. For example, a type argument can only be a `BaseType` and never a reference. Therefore, `BaseType` is preferred over `Type` for that struct field/function argument.